game-core/src/main/java/games/strategy/engine/auto/update/UpdatedMapsCheck.java

```diff
@@ -49,111 +44,52 @@ public static void checkDownloadedMapsAreLatest() {
-  }
-
-  private static Optional<Integer> readVersionFromPropertyFile(final File propertyFile) {
-    return DownloadFileProperties.loadForZipPropertyFile(propertyFile).getVersion();
-  }
-
-  private static Map<String, Integer> downloadAvailableMapsListAndComputeAvailableVersions() {
-    try {
-      return MapListingFetcher.getMapDownloadList().stream()
-          .collect(
-              Collectors.toMap(
-                  downloadFileDescription -> normalizeName(downloadFileDescription.getMapName()),
-                  DownloadFileDescription::getVersion));
-    } catch (final Exception e) {
-      log.warn("Failed to getting list of most recent maps", e);
-      return Map.of();
-    }
-  }
-
-  @VisibleForTesting
-  static Collection<String> computeOutOfDateMaps(
-      final Map<String, Integer> installedMapVersions,
-      final Map<String, Integer> availableToDownloadMapVersions) {
+  /**
+   * Computes maps that are out of date.
+   *
+   * @param availableToDownloadMaps List of maps that are available for download.
+   * @param mapVersionLookup Function given a map name returns installed map version (or empty if
+   *     the map is not installed).
+   * @return Set of map names that are installed where the available version is greater than the
+   *     installed version.
+   */
+  public static Collection<String> computeOutOfDateMaps(
+      final Collection<DownloadFileDescription> availableToDownloadMaps,
+      final Function<String, Optional<Integer>> mapVersionLookup) {
 
     final Collection<String> outOfDateMapNames = new ArrayList<>();
 
-    // Loop over all available maps, check if we have that map present by comparing
-    // normalized names, if so, check versions and remember any that are out of date.
-    for (final Map.Entry<String, Integer> installedMap : installedMapVersions.entrySet()) {
-      final String installedMapName = normalizeName(installedMap.getKey());
-
-      for (final Map.Entry<String, Integer> availableMap :
-          availableToDownloadMapVersions.entrySet()) {
-        final String availableMapName = normalizeName(availableMap.getKey());
-        if (installedMapName.equals(availableMapName)) {
-
-          if (availableMap.getValue() > installedMap.getValue()) {
-            outOfDateMapNames.add(availableMap.getKey());
-          }
-          break;
-        }
-      }
+    // Loop over all available maps, check if we have that map present, its version,
+    // and remember any whose version is less than what is available.
+    for (final DownloadFileDescription availableMap : availableToDownloadMaps) {
+      mapVersionLookup
+          .apply(availableMap.getMapName())
+          .ifPresent(
+              installedVersion -> {
+                if (installedVersion < availableMap.getVersion()) {
+                  outOfDateMapNames.add(availableMap.getMapName());
+                }
+              });
     }
     return outOfDateMapNames;
```

Comment:
```java
return availableToDownloadMaps.stream()
  .filter(availableMap -> mapVersionLookup.apply(availableMap.getMapName()).orElse(Integer.MAX_VALUE) < availableMap.getVersion())
  .collect(Collectors.toList());
  .map(DownloadFileDescription::getMapName)
```

Not sure if that's better tbh. Maybe if the filter is moved to a helper method?

Reply:
Yeah, that filter is overwhelming. A helper method is additional indirection, so not a clear win necessarily. I think I'm on the fence, could argue it both ways.

I think the effort needs to be put into `final Function<String, Optional<Integer>> mapVersionLookup`, I'm not a fan of having such a generic function be passed into a public API, makes the usage of this method very difficult.